src/Middleware/StaticFiles/src/Infrastructure/SharedOptions.cs

```diff
@@ -42,5 +42,10 @@ public PathString RequestPath
         /// The file system used to locate resources
         /// </summary>
         public IFileProvider FileProvider { get; set; }
+
+        /// <summary>
+        /// Indicates whether to add trailing slash at the end of path
+        /// </summary>
+        public bool AppendTrailingSlash { get; set; } = true;
```

Comment:
You're trying to suppress the redirect, not the trailing slash, right? You still add the trailing slash internally.
RedirectToAppendTrailingSlash?

Reply:
Renamed in next commit